summary_workflows/quantification/quantification_dockers/q_metrics/compute_metrics.py

```diff
@@ -30,146 +38,187 @@ def main(args):
 
 
-def compute_metrics(participant_input, gold_standards_dir, challenge_ids, participant, community, out_path, windows, genome_path):
+def compute_metrics(infile, gold_standards_dir, challenge, participant, community, out_path, windows, genome_path):
 
     # define array that will hold the full set of assessment datasets
     all_assessments = []
     # define list with keywords for return type of dataframe from match_with_gt()
     all_return_df_types = ["all_GT", "union", "intersection"]
 
-    for challenge in challenge_ids:
-
-        # ID prefix for assessment objects
-        base_id = f"{community}:{challenge}:{participant}:"
-        # Dict to store metric names and corresponding variables + stderr (which is currently not computed and set to 0)
-        metrics = {}
-        # ground truth file
-        gold_standard = os.path.join(gold_standards_dir, challenge + ".bed")
-        # genome annotation file
-        genome_file = select_genome_file(challenge, genome_path)
-        # log to stdout
-        print(f"INFO: In challenge {challenge}. Using genome file: {genome_file}")
-        genome = matchPAS.load_genome(genome_file)
-
-        for window in windows:
-            # obtain dataframes with matched and unmatched sites
-            matched, only_PD, only_GT = matchPAS.match_with_gt(f_PD=participant_input, f_GT=gold_standard, window=window)
-
-            # METRIC: Expression unmatched sites
-            ####################################
-            # Key: exact name of metric as it appears in specification
-            metric_name = f"Sum_FP_TPM:{window}nt"
-            
-            # Get metric
-            if not only_PD.empty:
-                # total expression of non-matched PD sites
-                tpm_unmatched = (only_PD["score_p"]).sum()
+    # ID prefix for assessment objects
+    base_id = f"{community}:{challenge}:{participant}:"
+    # Dict to store metric names and corresponding variables + stderr (which is currently not computed and set to 0)
+    metrics = {}
+
+    # check for presence of participant input data
+    assert os.path.exists(infile), "Participant file not found, please check input data."
+
+    # ground truth file
+    gold_standard = os.path.join(gold_standards_dir, challenge + ".bed")
+    assert os.path.exists(gold_standard), "Ground truth file not found, please check input data."
+
+    # genome annotation file
+    genome_file = apa.select_genome_file(challenge, genome_path)
+    # log to stdout
+    print(f"INFO: In challenge {challenge}. Using genome file: {genome_file}")
+    genome = apa.load_genome(genome_file)
+    
+
+    for window in windows:
+
+        ## Get matching sites
+        matched = apa.bedtools_window(infile, gold_standard, window)
+
+        # Number of duplicated PD sites, i.e. PD sites that matched multiple GT sites
+        dPD_count = sum(matched.duplicated(['chrom_p', 'chromStart_p', 'chromEnd_p', 'strand_p'], keep="first")) 
```

Comment:
Yes, these are not needed in the quantification workflow. Will be removed here as soon as the identification wf is set up. (In another PR, so please have mercy and ignore dPD_count and dGT_count)